client/components/data/domain-management/nameservers/index.jsx

```diff
@@ -1,80 +1,93 @@
-	}
-
 	return {
-		domains,
+		domains: props.domains,
 		nameservers: NameserversStore.getByDomainName( props.selectedDomainName ),
 		selectedDomainName: props.selectedDomainName,
-		selectedSite: props.selectedSite
+		// undefined is not permitted by the passed component
+		// so we have to explicitly choose a bool in that case
+		selectedSite: props.selectedSite || false,
 	};
 }
 
-const NameserversData = React.createClass( {
-	propTypes: {
-		component: React.PropTypes.func.isRequired,
-		selectedDomainName: React.PropTypes.string.isRequired,
-		sites: React.PropTypes.object.isRequired
-	},
+export class NameserversData extends Component {
+	static propTypes = {
+		component: PropTypes.func.isRequired,
+		selectedDomainName: PropTypes.string.isRequired,
+		selectedSite: PropTypes.object,
+	};
 
-	mixins: [ observe( 'sites' ) ],
+	constructor( props ) {
+		super( props );
 
-	componentWillMount() {
-		this.loadDomains();
 		this.loadNameservers();
-	},
+	}
 
 	componentWillUpdate() {
-		this.loadDomains();
 		this.loadNameservers();
-	},
-
-	loadDomains() {
-		const selectedSite = this.props.sites.getSelectedSite();
-
-		if ( this.prevSelectedSite !== selectedSite ) {
-			upgradesActions.fetchDomains( selectedSite.ID );
-
-			this.prevSelectedSite = selectedSite;
-		}
-	},
+	}
 
-	loadNameservers() {
-		upgradesActions.fetchNameservers( this.props.selectedDomainName );
-	},
+	loadNameservers = () => fetchNameservers( this.props.selectedDomainName );
 
 	render() {
+		const { selectedSite } = this.props;
+
 		return (
-			<StoreConnection
-				component={ this.props.component }
-				stores={ stores }
-				getStateFromStores={ getStateFromStores }
-				selectedDomainName={ this.props.selectedDomainName }
-				selectedSite={ this.props.sites.getSelectedSite() } />
+			<div>
+				<QuerySiteDomains siteId={ selectedSite && selectedSite.ID } />
+				<StoreConnection
+					component={ this.props.component }
+					domains={ this.props.domains }
+					stores={ stores }
+					getStateFromStores={ getStateFromStores }
+					selectedDomainName={ this.props.selectedDomainName }
+					selectedSite={ selectedSite }
+				/>
+			</div>
 		);
 	}
-} );
+}
+
+const mapStateToProps = state => {
+	const selectedSite = getSelectedSite( state );
+	const domains = getDomainsBySite( state, selectedSite );
+
+	return {
+		domains: {
+			// this doesn't appear to be currently
+			// stored in our app state as it was in
+			// the domains store. we know if we are
+			// currently fetching, but not if a fetch
+			// has taken place. in this case, I'm
+			// using the length of the array to indicate
+			// `hasLoaded` because this component should
+			// never exist in the app without at least
+			// one domain. thus, if we have at least one
+			// then we have loaded them.
+			hasLoadedFromServer: !! domains.length,
+			list: domains,
+		},
+		selectedSite
+	};
+};
 
-export default NameserversData;
+export default connect( mapStateToProps )( NameserversData );
```

Comment:
We export both default an a named class here. Old require imports would break:

```
const nameservers = require( 'components/data/domain-management/nameservers' )
```

vs

```
const nameservers = require( 'components/data/domain-management/nameservers' ).default;
```